terraform/azure/main.tf

```diff
@@ -0,0 +1,164 @@
+
+terraform {
+  backend "gcs" {
+    bucket = "two-eye-two-see-org-terraform-state"
+    prefix = "terraform/state/pilot-hubs"
+  }
+}
+provider "azurerm" {
+  subscription_id = var.subscription_id
+  features {}
+}
+
+resource "azurerm_resource_group" "jupyterhub" {
+  name     = var.resourcegroup_name
+  location = var.location
+}
+
+resource "azurerm_virtual_network" "jupyterhub" {
+  name                = "k8s-network"
+  location            = azurerm_resource_group.jupyterhub.location
+  resource_group_name = azurerm_resource_group.jupyterhub.name
+  address_space       = ["10.0.0.0/8"]
+}
+
+resource "azurerm_subnet" "node_subnet" {
+  name                 = "k8s-nodes-subnet"
+  virtual_network_name = azurerm_virtual_network.jupyterhub.name
+  resource_group_name  = azurerm_resource_group.jupyterhub.name
+  address_prefixes     = ["10.1.0.0/16"]
+}
+
+resource "azurerm_kubernetes_cluster" "jupyterhub" {
+  name                = "hub-cluster"
+  location            = azurerm_resource_group.jupyterhub.location
+  resource_group_name = azurerm_resource_group.jupyterhub.name
+  kubernetes_version  = var.kubernetes_version
+  dns_prefix          = "k8s"
+
+  linux_profile {
+    admin_username = "hub-admin"
+    ssh_key {
+      key_data = var.ssh_pub_key
+    }
+  }
+
+  # Core node-pool
+  default_node_pool {
+    name       = "core"
+    node_count = 1
+    # Unfortunately, changing anything about VM type / size recreates *whole cluster
+    vm_size             = var.core_node_vm_size
+    os_disk_size_gb     = 40
+    enable_auto_scaling = true
+    min_count           = 1
+    max_count           = 10
+    vnet_subnet_id      = azurerm_subnet.node_subnet.id
+    node_labels = {
+      "hub.jupyter.org/node-purpose" = "core",
+      "k8s.dask.org/node-purpose"    = "core"
+    }
+
+    orchestrator_version = var.kubernetes_version
+  }
+
+  auto_scaler_profile {
+    skip_nodes_with_local_storage = true
+  }
+
+  identity {
+    type = "SystemAssigned"
+  }
+
+  network_profile {
+    # I don't trust Azure CNI
```

Comment:
Good instinct - we found a bug on the Turing mybinder deployment that meant network policies were not being applied as expected!